pkg/cmd/cli/config/store.go

```diff
@@ -116,13 +116,15 @@ func (c *ConfigStore) SaveToFile(credentialsName string, namespace string, clien
 
 	credentials := clientcmdapi.NewAuthInfo()
 	credentials.Token = clientCfg.BearerToken
-	credentials.ClientCertificate = clientCfg.TLSClientConfig.CertFile
-	if len(credentials.ClientCertificate) == 0 {
-		credentials.ClientCertificateData = clientCfg.TLSClientConfig.CertData
-	}
-	credentials.ClientKey = clientCfg.TLSClientConfig.KeyFile
-	if len(credentials.ClientKey) == 0 {
-		credentials.ClientKeyData = clientCfg.TLSClientConfig.KeyData
+	if len(credentials.Token) == 0 {
```

Comment:
Fixed. I reverted this and now I'm making sure login does not not carry cert info from previous contexts. Still supported if explicitly provided.